library/src/scala/Tuple.scala

```diff
@@ -54,6 +55,13 @@ sealed trait Tuple extends Product {
   inline def map[F[_]](f: [t] => t => F[t]): Map[this.type, F] =
     runtime.Tuples.map(this, f).asInstanceOf[Map[this.type, F]]
 
+  /** Called on a tuple `(a1, ..., an)`, returns a new tuple `f(a1) ++ ... ++ f(an)`.
+   *  The result is typed as `Concat[F[A1], ...Concat[F[An], EmptyTuple]...])` if the tuple type is fully known.
+   */
+  @experimental
+  inline def flatMap[This >: this.type <: Tuple, F[_] <: Tuple](f: [t] => t => F[t]): FlatMap[This, F] =
```

Comment:
This signature is not ideal as we need to pass the tuple type when we call. This should always be inferred.
Maybe we should wait until we can do something like
```scala
  inline def flatMap[F[_] <: Tuple][This >: this.type <: Tuple](f: [t] => t => F[t]): FlatMap[This, F] =
```